netbox/templates/htmx/delete_form.html

```diff
@@ -12,6 +12,18 @@ <h5 class="modal-title">{% trans "Confirm Deletion" %}</h5>
         Are you sure you want to <strong class="text-danger">delete</strong> {{ object_type }} <strong>{{ object }}</strong>?
       {% endblocktrans %}
     </p>
+    {% if related_objects %}
+    <p>
+      This will cause deletion of the following items: <br>
+      {% for model,objects in related_objects.items %} 
+        - {{ model }}:
+        {% for object in objects %}
+          {{ object | linkify }}{% if not forloop.last %}, {% endif %}
+        {% endfor %}
+        <br>
+      {% endfor %}
```

Comment:
IMO we should present the list of dependent objects in a scrollable list. Bootstrap's [accordion component](https://getbootstrap.com/docs/5.1/components/accordion/) could work very well here.

Reply:
<img width="535" alt="Screenshot 2023-10-31 at 21 14 27" src="https://github.com/netbox-community/netbox/assets/1613241/551e9920-02ed-4641-9101-b3bc158c08b4">
<img width="533" alt="Screenshot 2023-10-31 at 21 14 32" src="https://github.com/netbox-community/netbox/assets/1613241/14510898-0f8b-4c62-9cb6-32492051b83d">

happier with this? (im using verbose_name and verbose_name_plural as needed for the accordion headers, some are capitalised, some are not)